kernel/cond.c

```diff
@@ -216,6 +252,7 @@ int myst_cond_requeue(
         for (myst_thread_t* p = requeues.front; p; p = next)
         {
             next = p->qnext;
+            p->signal.cond_wait = c2;
```

Comment:
Before your fix, requeued threads had their `cond_wait=NULL`?

Reply:
The requeued threads have `cond_wait` to be `c1` without my fix.